synthetics/test_conversions.go

```diff
@@ -143,11 +143,41 @@ func testURLToMapSlice(obj *synthetics.V202101beta1UrlTest) []map[string]interfa
 	}}
 }
 
+//nolint: gocyclo, gocognit
 func testHealthSettingsToMapSlice(obj *synthetics.V202101beta1HealthSettings) []map[string]interface{} {
 	if obj == nil {
 		return nil
 	}
 
+	if obj.LatencyCritical == nil &&
+		obj.LatencyWarning == nil &&
+		obj.PacketLossCritical == nil &&
+		obj.PacketLossWarning == nil &&
+		obj.JitterCritical == nil &&
+		obj.JitterWarning == nil &&
+		obj.HttpLatencyCritical == nil &&
+		obj.HttpLatencyWarning == nil &&
+		obj.HttpValidCodes == nil &&
+		obj.DnsValidCodes == nil {
+		return nil
+	}
+
+	// Kentik API sets 0 values for nil fields
```

Comment:
```suggestion
	// Kentik API sets 0 values for omitted optional fields.
```
I would avoid using _nil_ while describing JSON payloads.

Reply:
Addressed: 0e8ee0d8730ca72fcd78be1b5f7aa7a7f5c0e604